fix: add transparent wrapper to fix boolean Vue quirk

KammererTob opened the pull request:
This fixes the global/form-wide HTML validation disable configuration, by making sure that Vue treats non-explicit & optional booleans as undefined instead of casting to false.
Previously the most fine-grained control (i.e. setting it via useTextField props) would always overwrite the more global control, even if the boolean was never explicitly provided at that level.
It now uses the wrapper type `TransparentWrapper` instead of boolean directly, so that Vue correctly uses undefined for values that are not provided. The `toBooleanValue` function is needed to still support the property shorthand in templates (i.e. just providing `disable-html-validation`).

Supersedes: https://github.com/formwerkjs/formwerk/pull/198 and https://github.com/formwerkjs/formwerk/pull/199
Fixes: https://github.com/formwerkjs/formwerk/issues/197

Note: This fixes the build on Windows (at least on my machine) and also changed one deprecated function call in Zod test. I had to disable the one Zod test, to be able to push, because of an issue upstream https://github.com/colinhacks/zod/issues/5091

logaretm commented on packages/core/src/useSwitch/useSwitch.ts:
```diff
@@ -23,6 +23,7 @@ import { useFormField, exposeField } from '../useFormField';
 import { FieldTypePrefixes } from '../constants';
 import { useInputValidity } from '../validation';
 import { registerField } from '@formwerk/devtools';
+import { TransparentWrapper } from 'packages/core/dist/core';
```
```suggestion
import { TransparentWrapper } from '../../types';
```

logaretm commented on packages/core/src/useTextField/useTextField.ts:
```diff
@@ -15,6 +15,7 @@ import { useLabel, useErrorMessage } from '../a11y';
 import { useFormField, exposeField } from '../useFormField';
 import { FieldTypePrefixes } from '../constants';
 import { StandardSchema } from '../types';
+import { TransparentWrapper } from 'packages/core/dist/core';
```
```suggestion
import { TransparentWrapper } from '../../types';
```

logaretm commented on packages/core/src/validation/useInputValidity.ts:
```diff
@@ -7,12 +7,13 @@ import { isInputElement, normalizeArrayable, warn } from '../utils/common';
 import { FormGroupContext, FormGroupKey } from '../useFormGroup';
 import { getConfig } from '../config';
 import { checkLocaleMismatch } from '../i18n';
+import { TransparentWrapper } from 'packages/core/dist/core';
```
```suggestion
import { TransparentWrapper } from '../../types';
```

logaretm commented on packages/ecosystem/src/zod/zod.spec.ts:
```diff
@@ -2,6 +2,7 @@ import { fireEvent, render, screen } from '@testing-library/vue';
 import { z } from 'zod';
 import { useForm } from '@formwerk/core';
 import { flush } from '@test-utils/index';
+import { skip } from 'node:test';
```
we don't use that module, use `test.skip`

logaretm commented on packages/ecosystem/src/zod/zod.spec.ts:
```diff
@@ -45,10 +46,10 @@ test('zod schemas are supported', async () => {
 });
 
-test('collects multiple errors per field', async () => {
+skip('collects multiple errors per field', async () => {
```
use `test.skip`, but moreover why do we need to skip tests here?

logaretm replied:
Ah sorry, checked your PR description again and it makes sense. Just use `test.skip`

logaretm commented on scripts/config.ts:
```diff
@@ -42,7 +42,7 @@ async function createConfig(pkg: keyof typeof pkgNameMap, format: ModuleFormat)
     input: {
       resolve: {
-        tsconfigFilename: normalizePath(path.resolve(__dirname, `../tsconfig.lib.json`)),
+        tsconfigFilename: slashes(path.resolve(__dirname, `../tsconfig.lib.json`)),
```
is this a windows thing? if so let's take it out in another PR and I can merge that ASAP.

logaretm replied:
Read your description again, makes sense to have this but let's put it in another PR.

KammererTob replied:
I've removed this change from this PR

logaretm reviewed (changes requested):
Fantastic work, I think it looks good but there are a few minor gotchas with the import paths.

I left some comments, we can merge once addressed. Thanks a lot!

KammererTob commented:
Thanks for the review. I've fixed all issues you commented on. I have adjusted the changeset to be patch instead of minor. Or do is need to generate a new changeset for that?
